nagios/check.py

```diff
@@ -172,6 +176,27 @@ def check(self, instance):
             raise Exception('No Nagios configuration file specified')
         for tailer in self.nagios_tails[instance_key]:
             tailer.check()
+        i_key = {"type": self.INSTANCE_TYPE, "url": instance.get("nagios_conf")}
+        self.get_topology(i_key)
+
+    def get_topology(self, instance_key):
```

Comment:
can we add a test that mocks the Model.Host.objects.all payload and where we check whether we obtain/derive the correct components that would be sent to StackState? e.g. checking whether the right name is used etc, basic unit test

Reply:
Added the test case for the topology part to send components.